utils/gyb_syntax_support/ExprNodes.py

```diff
@@ -297,27 +288,20 @@
              Child("SecondChoice", kind='Expr')
          ]),
 
-    # a.b
+    # expr?.name
     Node('MemberAccessExpr', kind='Expr',
          children=[
              # The base needs to be optional to parse expressions in key paths
              # like \.a
              Child("Base", kind='Expr', is_optional=True),
-             Child("Dot", kind='PeriodToken'),
-             Child("Name", kind='Token'),
-             Child('DeclNameArguments', kind='DeclNameArguments',
-                   is_optional=True),
-         ]),
-
-    # dot-self-expr -> expr? '.' 'self'
-    Node('DotSelfExpr', kind='Expr',
-         children=[
-             Child('Expression', kind='Expr', is_optional=True),
-             Child('Dot', kind='Token',
+             Child("Dot", kind='Token',
                    token_choices=[
                        'PeriodToken', 'PrefixPeriodToken'
                    ]),
-             Child('SelfKeyword', kind='SelfToken'),
+             ## Name could be 'self'
+             Child("Name", kind='Token'),
```

Comment:
How do you distinguish `foo.self` and `` foo.`self` ``?

```code
struct Foo {
  var `self`: Int { return 0 }
}

let f = Foo()
print(f.self)
print(f.`self`)
```

Reply:
(I mean, yes, hopefully no one does this, but the compiler and language support it.)

Reply:
`self` currently parsed as one token so the structure still applies. It parsed to be something like: `<MemberAccessExpr><IdentifierExpr>f</IdentifierExpr>.`self`</MemberAccessExpr>`

Reply:
I see, `` `self` `` and `self` are different tokens, even though they have the same identifier text.